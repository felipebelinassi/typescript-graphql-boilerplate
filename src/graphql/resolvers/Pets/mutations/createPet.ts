import { Resolver, Query, Mutation, Arg } from 'type-graphql';
import { InjectRepository } from 'typeorm-typedi-extensions';
import Pets from '../types/objects/Pets';
import PetsRepository from '../../../../entities/Pets/PetsRepository';
import CreatePetInput from '../types/inputs/CreatePet';

@Resolver()
export default class CreatePetResolver {
  @InjectRepository(PetsRepository)
  private petsRepository!: PetsRepository;

  @Mutation(() => Pets, { description: 'Create a new pet in the database' })
  async createPet(
    @Arg('input', () => CreatePetInput)
    input: CreatePetInput
  ): Promise<Pets> {
    return this.petsRepository.createPet(input);
  }
}
