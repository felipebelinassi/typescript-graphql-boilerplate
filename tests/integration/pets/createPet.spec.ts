import { Connection } from 'typeorm';
import { createTestClient } from 'apollo-server-testing';
import { createPetMutation } from './__gql__/mutations';
import {
  createTestConnection,
  createTestServer,
  clearTestDb,
} from '../../utils';

let testConnection: Connection;

beforeAll(async () => {
  testConnection = await createTestConnection();
});

beforeEach(async () => {
  await clearTestDb(testConnection);
});

afterAll(async () => {
  await testConnection.close();
});

describe('Create Pets', () => {
  it('should register a new pet in the database', async () => {
    const petInput = {
      name: 'Test dog',
      breed: 'Tester',
      species: 'Canine',
      birthDate: '2021-05-07',
    };

    const apolloServer = await createTestServer();
    const { query } = createTestClient(apolloServer);

    const res = await query({
      query: createPetMutation,
      variables: petInput,
    });
    expect(res).toMatchSnapshot('New pet');
  });
});
