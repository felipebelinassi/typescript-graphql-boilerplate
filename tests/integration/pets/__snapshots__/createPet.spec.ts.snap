// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Pets should register a new pet in the database: New pet 1`] = `
Object {
  "data": Object {
    "createPet": Object {
      "birthDate": "2021-05-07",
      "breed": "Tester",
      "id": 1,
      "name": "Test dog",
      "species": "Canine",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
